activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -19,6 +19,11 @@ module AttributeMethods
       include TimeZoneConversion
       include Dirty
       include Serialization
+
+      # Provides access to the value of the record's id column.
+      # Use +uid+ over +id+ if the raw id column value is needed,
+      # rather than the value of the primary key.
+      alias_attribute :uid, :id
```

Comment:
Should we consider making it conditional to avoid generate methods on models that don't have `id` column/attribute? 
I think regardless of our choice we may want to have a test for a model that doesn't have `id` column in the schema just to see what happens in this case and basically serve as explicit demonstration of the expected behavior

```suggestion
      alias_attribute :uid, :id if has_attribute?(:id)
```

Reply:
Yep, that makes sense to me! In terms of correctness, it should be fine to use `alias_attribute` for a model without an `id` column, because the method will get aliased to the `attribute(:id)` method and just return `nil`. But might as well avoid defining a bunch of aliased methods if the model is missing an `:id` attribute in the first place.

I can add a test with a model whose table doesn't include `id` 👍

Reply:
Hm, actually I don't think we can use `has_attribute?(:id)` -- it calls `#reset_table_name`, which won't work in the context of `ActiveRecord::Base` (on which we `include AttributeMethods`) because `superclass` is `Object`, so `superclass.abstract_class?` doesn't work:
https://github.com/rails/rails/blob/48243d42aad9675e26408aaa561e677dac91db04/activerecord/lib/active_record/model_schema.rb#L279-L284

I can still write a test that ensures that calling `uid` on a model without an id columns works as expected, but seems we're essentially just testing that `attribute(:id)` returns nil, I'm not sure it's worth it. What do you think?

Reply:
I didn't realize we were in the context of `ActiveRecord::Base`, I was thinking we were in a concrete model's context. Actually I'm not sure if it's good or bad. It's good since we will only generate attribute methods once and these methods will live in `ActiveRecord::Base` so technically we won't be generating methods on models that won't be using it. 

`uid` returning `nil` seems fine to me. It's a little inconsistent with let's say how `object.non_existing_attribute` will behave but I don't see it as a concern. 

Definitely not a question for this PR but I wonder if in the future we could allow `uid` to be configurable for models that don't necessarily have the `id` column but want to implement the `uid` interface. 
So later we could see `alias_attribute :uid, uid_attribute_source_column` where `uid_attribute_source_column` defaults to `:id` but other models can override it if needed.

@rafaelfranca Is is what you imagined for the `uid` attribute? It is going to be a method to access a single-value identifier of an object regardless of what is being used as the `primary_key`, if a model has such concept